packages/shared/src/components/post/freeform/write/WritePage.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { ReactElement } from 'react';
+import { WritePageContainer } from './common';
+import { WritePostHeader } from './WritePostHeader';
+import {
+  WriteFreeformContent,
+  WriteFreeformContentProps,
+} from './WriteFreeformContent';
+import { WriteFreeFormSkeleton } from './WriteFreeFormSkeleton';
+import { verifyPermission } from '../../../../graphql/squads';
+import { SourcePermissions, Squad } from '../../../../graphql/sources';
+import Unauthorized from '../../../errors/Unauthorized';
+
+interface WritePageProps extends Omit<WriteFreeformContentProps, 'squadId'> {
+  isEdit?: boolean;
+  isLoading?: boolean;
+  isForbidden?: boolean;
+  squad: Squad;
+}
+
+export function WritePage({
+  isEdit,
+  squad,
+  isLoading,
+  isForbidden,
+  ...props
+}: WritePageProps): ReactElement {
+  const isVerified = verifyPermission(squad, SourcePermissions.Post);
+
+  if (isLoading) return <WriteFreeFormSkeleton isEdit={isEdit} />;
+
+  if (isForbidden || !isVerified || !squad) return <Unauthorized />;
```

Comment:
Looks to me that we could just return `Unauthorized` first without waiting for loading since all info should be there squad, post, user?

So just flipping two ifs.

Reply:
We need to show the loading first, we don't fetch the static props for this page, so we won't have the proper information.